lib/view_model/active_record/update_data.rb

```diff
@@ -55,14 +160,158 @@ def initialize(values)
     end
 
     class Replace < self
-      def update_datas
+      def update_datas(referenced_updates)
         values
       end
     end
 
     class Functional < self
-      def update_datas
-        values.flat_map(&:values)
+      def update_datas(referenced_updates)
+        values.reject { |action| action.is_a?(FunctionalUpdate::Remove) }.flat_map(&:values)
+      end
+    end
+
+    class Parser < UpdateParser
+      attr_reader :parse_association
+
+      def initialize(association_data, blame_reference, parse_association:)
+        super(association_data, blame_reference)
+        @parse_association = parse_association
+      end
+
+      def action_schema
+        UpdateData::Schemas::COLLECTION_UPDATE
+      end
+
+      def parse_update_values(values)
```

Comment:
Ah, I see. I didn't actually understand at the time, I thought it was inside one of the fupdate classes, shadowing the `values` member. Mistake, but still a new name would prevent that confusion.